### PR TITLE
vai pagination tests - tokens

### PR DESCRIPTION
**Pull request by yonasberhe23**
### Summary
Fixes #https://github.com/rancher/qa-tasks/issues/1380

- Covers pagination functionality for tokens list under Account & API Keys.
- Refactor pagination tests for events

### Checklist
- [x] The PR is linked to an issue and the linked issue has a Milestone, or no issue is needed
- [x] The PR has a Milestone 
- [x] The PR template has been filled out
- [x] The PR has been self reviewed 
- [x] The PR has a reviewer assigned
- [x] The PR has automated tests or clear instructions for manual tests and the linked issue has appropriate QA labels, or tests are not needed
- [ ] The PR has reviewed with UX and tested in light and dark mode, or there are no UX changes

### CODE REVIEW

File: cypress/e2e/tests/pages/explorer/workloads/pods.spec.ts [hunk 1]
```diff
@@ -7,41 +7,50 @@ import { generatePodsDataSmall } from '@/cypress/e2e/blueprints/explorer/workloa
...
+
   describe('List', { tags: ['@vai'] }, () => {
     before('set up', () => {
-      cy.updateResourceListViewPref('local', 'none', '{\"local\":[]}');
+      cy.updateNamespaceFilter('local', 'none', '{\"local\":[]}');
 
-      // create namespaces
-      cy.createNamespace(nsName);
-      cy.createNamespace(nsName2);
+      cy.createE2EResourceName('namespace1').then((nsName1) => {
```
Comment (richard-cox) [inline cypress/e2e/tests/pages/explorer/workloads/pods.spec.ts:26]: this and below have already been created in the beforeEach

File: cypress/e2e/tests/pages/explorer/dashboard/events.spec.ts [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+  beforeEach(() => {
+    cy.createE2EResourceName('namespace1').as('namespaceName1');
+    cy.createE2EResourceName('namespace2').as('namespaceName2');
+  });
+
+  describe('List', { tags: ['@vai'] }, () => {
+    before('set up', () => {
+      cy.updateNamespaceFilter('local', 'none', '{\"local\":[]}');
+
+      cy.createE2EResourceName('namespace1').then((nsName1) => {
```
Comment (richard-cox) [inline cypress/e2e/tests/pages/explorer/dashboard/events.spec.ts:26]: same here, namespace1 and namespace2 have already been created in beforeEach

File: cypress/e2e/tests/pages/explorer/dashboard/events.spec.ts [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+        // create namespace
+        cy.createNamespace(nsName1);
+
+        // create pods
+        let i = 0;
+
+        while (i < 125) {
+          const podName = `e2e-${ Cypress._.uniqueId(Date.now().toString()) }`;
+
+          cy.createPod(nsName1, podName, 'nginx:latest', false).then(() => {
```
Comment (richard-cox) [inline cypress/e2e/tests/pages/explorer/dashboard/events.spec.ts:36]: is this done to create the large number of events?

**Review by richard-cox — CHANGES_REQUESTED** [review]

File: cypress/e2e/tests/pages/explorer/dashboard/events.spec.ts (hunk 3 above)
Comment (yonasberhe23) [reply cypress/e2e/tests/pages/explorer/dashboard/events.spec.ts:36]: correct

File: cypress/e2e/tests/pages/explorer/workloads/pods.spec.ts (hunk 1 above)
Comment (yonasberhe23) [reply cypress/e2e/tests/pages/explorer/workloads/pods.spec.ts:26]: correct, but the before() hooks run prior to the beforeEach() so when we only create it in the beforeEach() an error is thrown (e.g. `namespace1` is not found). To resolve this, I'll declare the variable prior to the before() and get rid of the beforeEach() hook.

File: cypress/e2e/tests/pages/explorer/dashboard/events.spec.ts (hunk 2 above)
Comment (yonasberhe23) [reply cypress/e2e/tests/pages/explorer/dashboard/events.spec.ts]: see above comment

**Review by richard-cox — APPROVED** [review]

**Pull request merged**